File: ros_introduction/src/ros_unity_robotics/src/pos_pub.cpp
#include "ros/ros.h"
#include "std_msgs/Float32.h"

float POSITION = 0.3;

int main(int argc, char **argv)
{
  ros::init(argc, argv, "position_pub_node");
  ros::NodeHandle node_handle;
  ros::Publisher pos_pub = node_handle.advertise<std_msgs::Float32>("position", 10);
  ros::Rate pub_rate(10);

  ROS_INFO("Publishing Position...");
  while(ros::ok())
  {
    std_msgs::Float32 msg;

    msg.data = POSITION + 0.025;
    POSITION = msg.data;

    pos_pub.publish(msg);
    pub_rate.sleep();
  }
}

File: ros_introduction/src/ros_unity_robotics/scripts/pos_pub.py
#!/usr/bin/env python2.7

import rospy
from std_msgs.msg import Float32

def pos_pub():
    rospy.init_node("position_pub_node")
    pub = rospy.Publisher("position", Float32, queue_size=10)

    print("Publishing Position...")
    pos = 1

    rate = rospy.Rate(5)
    while not rospy.is_shutdown():
        new_pos = pos + 0.025
        pos = new_pos
        pub.publish(pos)
        rate.sleep()

if __name__ == '__main__':
    try:
        pos_pub()

    except rospy.ROSInterruptException:
        pass
